[android] kotlin version for react-native 0.71

<!--- Please fill out the template to the best of your ability -->

With react-native 0.71 because of new version of gradle plugin we have next error on build project

```
FAILURE: Build failed with an exception.

* What went wrong:
The Android Gradle plugin supports only kotlin-android-extensions Gradle plugin version 1.6.20 and higher.
The following dependencies do not satisfy the required version:
project ':amplitude_analytics-react-native' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.5.30

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 8s
```

## Expected Behavior
All libraries should get rootProject version of `koltinVersion` instead hardcode

## Current Behavior
`android/build.gradle`
```
buildscript {
    ext.kotlinVersion = '1.5.30'
...
```

## Possible Solution
```
buildscript {
    ext.kotlinVersion = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.6.20"
...
```

## Steps to Reproduce
No applicable

## Environment
No applicable

--- a/packages/analytics-react-native/android/build.gradle
+++ b/packages/analytics-react-native/android/build.gradle
@@ -1,5 +1,5 @@
 buildscript {
-    ext.kotlinVersion = "1.5.30"
+    ext.kotlinVersion = rootProject.ext.has('kotlinVersion') ? rootProject.ext.get('kotlinVersion') : "1.5.30"
     repositories {
         google()
         jcenter()

